controllers/etcd_controller.go

```diff
@@ -1123,11 +1197,111 @@ func (r *EtcdReconciler) waitUntilStatefulSetReady(sts *appsv1.StatefulSet) (*ap
 		if err := health.CheckStatefulSet(ss); err != nil {
 			return gardenerretry.MinorError(err)
 		}
+		if ss.Status.Replicas != *ss.Spec.Replicas {
+			return gardenerretry.MinorError(fmt.Errorf("statefulset not scaled"))
+		}
 		return gardenerretry.Ok()
 	})
 	return ss, err
 }
 
+func (r *EtcdReconciler) getStatefulSet(ctx context.Context, etcd *druidv1alpha1.Etcd) (*appsv1.StatefulSet, error) {
```

Comment:
How does this interact with `claimStatefulSets`?

Reply:
Can we not avoid this func? Ideally, resequencing the reconcile/delete flow can get us `reconcileStatefulSet` before PVC clean up and we could get the statetulset from there. For snapshot we do not need the statefulset or PVCs anyway as mentioned above.